### PR TITLE
Charlesmchen/media gallery cells

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
A first draft of the changes to render media galleries in conversation view.

This is pretty incomplete; I added a lot of TODOs.

PTAL @michaelkirk 

img_0139
img_0138

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        case 0:
+            return
+        case 1:
+            guard let itemView = itemViews.first else {
+                owsFailDebug("Missing item view.")
+                return
+            }
+            addSubview(itemView)
+            itemView.autoPinEdgesToSuperviewEdges()
+        case 4:
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift]: no biggie, but is there a reason our case statements aren't increasing monotonically?

File: Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+            let rightViews = Array(itemViews[1..<3])
+            addArrangedSubview(newRow(rowViews: rightViews,
+                                      axis: .vertical,
+                                      viewSize: smallImageSize))
+            self.axis = .horizontal
+            self.spacing = MediaGalleryCellView.kSpacingPts
+        default:
+            // X X
+            // xxx
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift]: Small thing, but the ascii diagrams are helpful - thanks.

File: Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+        backgroundColor = Theme.backgroundColor
+
+        createContents(maxMessageWidth: maxMessageWidth)
+    }
+
+    private func createContents(maxMessageWidth: CGFloat) {
+        switch itemViews.count {
+        case 0:
+            return
```
[the remark this answers is not in the document]
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift]: Done.  We're waiting on related design decisions about whether and how "downloading" and "invalid" items are displayed.

File: Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift]: Done.  I was trying to align with how the (very similar) measurement logic is laid out, and the 1 and 4 cases are treated identically there.

**Pull request merged**